src/scripts/build_nst_da.py

```diff
@@ -0,0 +1,332 @@
+            case "train_audio":
+                data_dir = Path(name) / "dk"
+                for audio_dir in data_dir.iterdir():
+                    if not audio_dir.is_dir():
+                        continue
+                    for audio_file in audio_dir.glob("*.wav"):
+                        Path(audio_file).rename(train_audio_dir / audio_file.name)
+                shutil.rmtree(name)
+
+            # This file contains the test set as well as some corrections of errors in
+            # the training dataset
+            case "test_and_errors":
+                data_dir = Path(name) / "supplement_dk"
+
+                temp_test_audio_dir = data_dir / "testdata" / "audio"
+                for audio_dir in temp_test_audio_dir.iterdir():
+                    if not audio_dir.is_dir():
+                        continue
+                    for audio_file in audio_dir.glob("*.wav"):
+                        Path(audio_file).rename(test_audio_dir / audio_file.name)
+                temp_test_metadata_dir = data_dir / "testdata" / "metadata"
+                shutil.move(temp_test_metadata_dir, test_dir)
+
+                test_log_file = data_dir / "testdata" / "sprakbanken_0611_transform.log"
+                Path(test_log_file).rename(data_dir / "test" / "log.log")
+
+                error_file = data_dir / "dk_errorfiles_train.json"
+                Path(error_file).rename(train_dir / "errorfiles.json")
+
+                test_manifest_file = data_dir / "dk_manifest_test.json"
+                Path(test_manifest_file).rename(test_dir / "manifest.json")
+
+                shutil.rmtree(name)
+            case "metadata_csvs":
+                train_metadata_csv = Path(name) / "NST_dk.csv"
+                Path(train_metadata_csv).rename(train_dir / "metadata.csv")
+                test_metadata_csv = Path(name) / "supplement_dk.csv"
+                Path(test_metadata_csv).rename(test_dir / "metadata.csv")
+                shutil.rmtree(name)
+            case "readme":
+                Path(name).rename("README.pdf")
+
+
+def remove_bad_files() -> None:
+    """Remove audio files that cannot be opened."""
+    # These filename prefixes were found by running the `find_faulty_audio_clips.py`
+    # script
+    bad_file_prefixes = [
+        "dk11x242-18072000-1149_u0047",
+        "dk16xx41-24092000-1951_u0042",
+    ]
+    for split in ["train", "test"]:
+        audio_dir = Path(split) / "audio"
+        for audio_file in audio_dir.glob("*.wav"):
+            if any(
+                [
+                    audio_file.stem.startswith(bad_prefix)
+                    for bad_prefix in bad_file_prefixes
+                ]
+            ):
+                logger.info(f"Removing {audio_file} as it cannot be opened.")
+                audio_file.unlink()
+                continue
+
+            # Small files are also bad, as they are likely to be empty
+            file_size = audio_file.stat().st_size
+            if file_size < 8192:
```

Comment:
Motivation behind this size choice?

Reply:
I wanted to remove empty files, which were registered as having size ~1000-2000 bytes. I chose 8192 (2^13) bytes just to have a bit of a buffer.